src/compas_cgal/straight_skeleton_2.py

```diff
@@ -155,6 +155,63 @@ def create_offset_polygons_2(points, offset) -> list[Polygon]:
     return [Polygon(points.tolist()) for points in offset_polygons]
 
 
+def create_offset_polygons_with_holes_2(points, holes, offset) -> list[Tuple[Polygon, list[Polygon]]]:
+    """Compute the polygon offset with holes.
+
+    Parameters
+    ----------
+    points : list of point coordinates or :class:`compas.geometry.Polygon`
+        The points of the polygon.
+    holes : list of list of point coordinates or list of :class:`compas.geometry.Polygon`
+        The holes of the polygon.
+    offset : float
+        The offset distance. If negative, the offset is outside the polygon, otherwise inside.
+
+    Returns
+    -------
+
+    Returns
+    -------
+    list of tuple of (:class:`Polygon`, list[:class:`Polygon`])
+        The polygons with holes.
+
+    Raises
+    ------
+    ValueError
+        If the normal of the polygon is not [0, 0, 1].
```

Comment:
should it be a unit normal, or vertical up?

Reply:
should be vertical up, i've adjusted the docstrings